server.js

```diff
@@ -0,0 +1,82 @@
+const net = require("net");
+const files = require("./bodytext.js");
+
+const PORT = 8080;
+
+const server = net.createServer(socket => {
+  socket.on("data", chunk => {
+    let arr = chunk.toString().split("\r\n");
+
+    let requestObj = {};
+
+    requestObj.URI = arr[0].split(" ");
+    requestObj.host = arr[1].slice(arr[1].indexOf(":") + 2);
+    requestObj.userAgent = arr[2].slice(arr[2].indexOf(" ") + 1);
+    requestObj.accept = arr[3].slice(arr[3].indexOf(" ") + 1);
```

Comment:
These headers may not always be in this order. It's best to take it line by line, split by the `:` and use the first value to try and see which header to set, and the second value as the header's value.